### ISSUE TITLE
About pytorch-3d-gan

### ISSUE THREAD

**Issue by shukuang-1** [pclubiitk/model-zoo#71]
In utils.py,
evalArray[evalArray>T]=True

evalArray[evalArray<T]=False
I don't know the 'T'.
When I run the code,T will cause error.(NamError).
and when I delete it,the picture will be a red cube,can't generate model what we need.
Can you tell me about it.Please.

**Comment by gurbaaz27** [comment]
@SomTambe please see this.

**Comment by SomTambe** [comment]
Hi @shukuang-1 !
What you observed is absolutely right. The error will occur.
T here is the threshold for displaying the voxel. Above this value, the voxel will be displayed. The ConvTranspose3d outputs a 3d cubic, on which sigmoid has been applied to reduce the values to a range of [0,1].
On a completely trained model, the value of T should be equal to 0.5. That should be about 50000 epochs of training.
But I suggest you may evaluate the model using a series of T, mainly varying from a range of 0.5 to 0.99. You might get some kind of a figure if T=0.9 at 3000 epochs I suppose, but most people who trained the model successfully trained it for about 50000 epochs. 
For lesser number of epochs, you may get a full red block since generative adversarial networks have been observed to learn 3-dimensional latent space representations very slowly, rather inefficiently. That is the reason for such a high number of epochs.
Thanks @gurbaaz27 for refering me to this issue!

**Comment by shukuang-1** [comment]
Thanks for your answer  @SomTambe.I have solve this question.And I get some result after 10000 epochs,but it's not good.I will try to train 50000 epochs.

**Comment by SomTambe** [comment]
I shall close this issue here. Kindly reopen if any problem occurs.
